ApiDoctor.Console/Auth/OAuthAccount.cs

```diff
@@ -203,9 +203,10 @@ private async Task RedeemUsernameAndPasswordAsync()
         {
             if (string.IsNullOrEmpty(this.AccessToken))
             {
-                var creds = new Microsoft.IdentityModel.Clients.ActiveDirectory.UserCredential(this.Username, this.Password);
                 var context = new Microsoft.IdentityModel.Clients.ActiveDirectory.AuthenticationContext(this.TokenService);
-                var token = await context.AcquireTokenAsync(this.Resource, this.ClientId, creds);
+                var deviceCodeFlow = await context.AcquireDeviceCodeAsync(this.Resource, this.ClientId);
```

Comment:
I don't expect that this will work as intended. Replacing the user credential flow with device code flow won't work as I expect that this is typically run without user interaction. This could be an option when run with a user.

Reply:
This really depends whether it was used for local testing or for some CI. @BarryShehadeh do you have more insights on that aspect?

Reply:
We do run it as both during functional tests.
Is AcquireTokenAsync no longer available?
I would avoid any logic changes in this PR until we get dotnet core out of the door.

Reply:
@MIchaelMainer
I know that my team use it for functional tests, and it's probably a breaking change. However, we are using an older version of API Doctor with inline dlls. 
 Are you aware of any other use case scenarios for this?

Reply:
@MIchaelMainer I didn't know this was used unattended. I replaced the whole thing by a call straight to the endpoint since MSAL removed the capability in an effort to discourage people from doing it.